tests/test.js

```diff
@@ -11,3 +11,33 @@ test('test require hook', t => {
   t.equals(result.data.meaningOfLife, 42);
   t.end();
 });
+
+test('test compile errors', t => {
+  require('../register.js');
+  let unsupported_path = '../example/unsupported.js';
+
+  try {
+    require(unsupported_path);
```

Comment:
@b0rza I don't know what you did but this thing indeed works. Both errors have been caught and test suite passes. I've tried both on node v4.5.0 and v6.6.0...